File: desktop_app/front-end/src/cryptography/cipher.ts
const XOREncryptDecrypt = (message: string, keyString: string) => {
	const key = keyString.split("");
	const output = [];
	for (let i = 0; i < message.length; i++) {
		const charCode =
			message.charCodeAt(i) ^ key[i % key.length].charCodeAt(0);
		output.push(String.fromCharCode(charCode));
	}
	return output.join("");
};

export const encrypt = (message: string, key: string) =>
	XOREncryptDecrypt(message, key);
export const decrypt = (encryptedMessage: string, key: string) =>
	XOREncryptDecrypt(encryptedMessage, key);

File: phone_app/lib/src/util/cipher.dart
String _xorEncryptDecrypt(String input, String key) {
  final List<String> keyList = key.split('');
  final List<String> output = [];
  for (int i = 0; i < input.length; i++) {
    final int charCode =
        input.codeUnitAt(i) ^ keyList[i % keyList.length].codeUnitAt(0);
    output.add(new String.fromCharCode(charCode));
  }
  return output.join("");
}

String encrypt(String message, String key) {
  return _xorEncryptDecrypt(message, key);
}

String decrypt(String encryptedMessage, String key) {
  return _xorEncryptDecrypt(encryptedMessage, key);
}
